install.sh

```diff
@@ -88,11 +81,24 @@ else
   exit 1
 fi
 
-LATEST_RELEASE="$(eval "${HTTP_GET} '${RELEASE_INFO_URL}'")"
+if [[ "${BUILDKITE_AGENT_VERSION:-"latest"}" == "latest" ]]; then
+    echo -e "Finding latest release..."
+
+    RELEASE_INFO_URL="https://buildkite.com/agent/releases/latest?platform=${PLATFORM}&arch=${ARCH}&system=${SYSTEM}&machine=${MACHINE}"
+    if [[ "${BETA:-}" == "true" ]]; then
+      RELEASE_INFO_URL="${RELEASE_INFO_URL}&prerelease=true"
+    fi
+
+    LATEST_RELEASE="$(eval "${HTTP_GET} '${RELEASE_INFO_URL}'")"
 
-VERSION="$(          echo "${LATEST_RELEASE}" | awk -F= '/version=/  { print $2 }')"
-DOWNLOAD_FILENAME="$(echo "${LATEST_RELEASE}" | awk -F= '/filename=/ { print $2 }')"
-DOWNLOAD_URL="$(     echo "${LATEST_RELEASE}" | awk -F= '/url=/      { print $2 }')"
+    VERSION="$(          echo "${LATEST_RELEASE}" | awk -F= '/version=/  { print $2 }')"
+    DOWNLOAD_FILENAME="$(echo "${LATEST_RELEASE}" | awk -F= '/filename=/ { print $2 }')"
+    DOWNLOAD_URL="$(     echo "${LATEST_RELEASE}" | awk -F= '/url=/      { print $2 }')"
+else
+    VERSION=$BUILDKITE_AGENT_VERSION
```

Comment:
(slightly less nitpicky) Quotes and curly braces ftw:
```suggestion
    VERSION="${BUILDKITE_AGENT_VERSION}"
```